### ISSUE TITLE
program runs out of memory on raspberry pi board

### ISSUE THREAD

**Issue by tbird20d** [tbird20d/bird-slideshow#26]
The program runs out of memory on the raspberry pi board.

The error is:

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "/opt/Python3.8.15/lib/python3.8/tkinter/__init__.py", line 1892, in __call__
    return self.func(*args)
  File "/opt/Python3.8.15/lib/python3.8/tkinter/__init__.py", line 814, in callit
    func(*args)
  File "./bird-slideshow.py", line 818, in next_img
    async_preload_img()
  File "./bird-slideshow.py", line 584, in async_preload_img
    slideshow_imgs[preload_index].load_pil_from_path()
  File "./bird-slideshow.py", line 168, in load_pil_from_path
    filepath = download_ssh_img(config.cache_dir, self.img_path)
  File "./bird-slideshow.py", line 675, in download_ssh_img
    p = subprocess.run(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
  File "/opt/Python3.8.15/lib/python3.8/subprocess.py", line 493, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/opt/Python3.8.15/lib/python3.8/subprocess.py", line 858, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "/opt/Python3.8.15/lib/python3.8/subprocess.py", line 1639, in _execute_child
    self.pid = _posixsubprocess.fork_exec(
OSError: [Errno 12] Cannot allocate memory
```

**Comment by tbird20d** [comment]
Notes:

My raspberry pi board only had 1G of RAM, of which only 485M is available.  As images are downloaded, the size of
the process gets bigger and bigger (because PIL images and tk images are not flushed when they are removed from
the screen).

It might be possible to dispose of PIL and tk image data when an image is replaced on the screen.

Note that /proc/{pid}/statm has memory usage data for a process.
/proc/self/statm should have that info for the python3 process running bird-slideshow.py
The resident set size is the second field in the statm file.

**Comment by tbird20d** [comment]
Fixed by addd5dee91eca14b1123d9889efb6e0dc7bee2eb
